src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -348,12 +394,158 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     }
                     else
                     {
-                        ApplyResourceInNestedProperty(edmProperty, resource, resourceWrapper, readContext);
+                        ODataResourceWrapper resWrapper = resourceWrapper;
+                        if (resourceWrapper.Resource.Id != null)
+                        {
+                            resWrapper = UpdateResourceWrapper(edmProperty.Type, resWrapper.Resource.Id, readContext);
+
+                        }
+                        ApplyResourceInNestedProperty(edmProperty, resource, resWrapper, readContext);
                     }
                 }
             }
         }
 
+
+        private ODataResourceSetWrapper CreateResourceSetWrapper(IEdmCollectionTypeReference edmPropertyType,
+    IList<ODataEntityReferenceLinkWrapper> refLinks, ODataDeserializerContext readContext)
+        {
+            ODataResourceSet resourceSet = new ODataResourceSet
+            {
+                TypeName = edmPropertyType.FullName(),
+            };
+
+            IEdmTypeReference elementType = edmPropertyType.ElementType();
+            ODataResourceSetWrapper resourceSetWrapper = new ODataResourceSetWrapper(resourceSet);
+            foreach (ODataEntityReferenceLinkWrapper refLinkWrapper in refLinks)
+            {
+                ODataResourceWrapper resourceWrapper = CreateResourceWrapper(elementType, refLinkWrapper, readContext);
+                resourceSetWrapper.Resources.Add(resourceWrapper);
+            }
+
+            return resourceSetWrapper;
+        }
+
+        private ODataResourceWrapper CreateResourceWrapper(IEdmTypeReference edmPropertyType, ODataEntityReferenceLinkWrapper refLink, ODataDeserializerContext readContext)
+        {
+            Contract.Assert(readContext != null);
+
+            ODataResource resource = new ODataResource
+            {
+                TypeName = edmPropertyType.FullName(),
+            };
+
+            Uri serviceRootUri = null;
+            string serviceRoot = "";
+            DefaultODataPathHandler pathHandler = (DefaultODataPathHandler)readContext.InternalRequest.PathHandler;
+            if (!refLink.EntityReferenceLink.Url.IsAbsoluteUri)
```

Comment:
From the spec, the referenceLink can be a relativeUri. Is that supported in this feature?

Reply:
@KenitoInc  I can't see any of this..I think you have not reviewed the latest files.. Also there is a test file.

Reply:
yes